cloud/linode/loadbalancers.go

```diff
@@ -485,26 +484,24 @@ func getNodeInternalIP(node *v1.Node) string {
 	return ""
 }
 
-func getTLSCertInfo(kubeClient kubernetes.Interface, tlsAnnotations []*tlsAnnotation, namespace string, port int) (string, string, error) {
-	for _, tlsAnnotation := range tlsAnnotations {
-		if tlsAnnotation.Port == port {
-			secret, err := kubeClient.CoreV1().Secrets(namespace).Get(tlsAnnotation.TLSSecretName, metav1.GetOptions{})
-			if err != nil {
-				return "", "", err
-			}
+func getTLSCertInfo(kubeClient kubernetes.Interface, namespace string, portConfig *portConfig) (string, string, error) {
+	if portConfig.TLSSecretName == "" {
+		return "", "", fmt.Errorf("cert & key for port %v is not specified", portConfig.Port)
```

Comment:
Can this be `TLS secret name for port %v is not specified`? The error itself is returned with the secret name isn't specified, while the error makes it seem as if the cert and key keys in the secret need to be specified, which is not the case.